src/jv_aux.c

```diff
@@ -500,7 +500,7 @@ jv jv_delpaths(jv object, jv paths) {
     jv_free(paths);
     return jv_invalid_with_msg(jv_string("Paths must be specified as an array"));
   }
-  paths = jv_sort(paths, jv_copy(paths));
+  paths = jv_sort(jv_array_slice(paths, 0, jv_array_length(jv_copy(paths))), jv_copy(paths));
```

Comment:
The order of execution of arguments in unspecified.
You need to use a separate variable to enforce order or evaluation, or if the compiler decides to call `jv_array_slice(path, ...)` before `jv_copy(path)`, `jv_copy()` will copy something that was already freed.

Reply:
Also I am not calling `jv_array_slice` here is fine; afaict it modifies the array inplace in some cases instead of returning a new array.

Reply:
@emanuele6 makes sense. I initially had this block, I think this would fix the potential issue, any thoughts?
```
jv paths_copy = jv_copy(paths);
jv paths_sliced = jv_array_slice(paths_copy, 0, jv_array_length(paths_copy));
jv sorted_paths = jv_sort(paths_sliced, paths_copy);
jv_free(paths);
paths = sorted_paths;
```

Reply:
Hmm, that is wrong, because you are using the same copy of `paths` (`paths_copy`) three times.
You just need something like this:
```c
  jv slice = jv_array_slice(jv_copy(paths), 0, jv_array_length(jv_copy(paths)))
  paths = jv_sort(slice, paths);
```

Reply:
I am not sure if you actually understood the problem, so I will try to explain it more clearly:

In C, if you have:
```c
  c = function(A(), B());
```
It is unspecified whether that gets executed as:
```c
  {
    jv a = A();
    jv b = B();
    c = function(a, b);
  }
```
or as:
```c
  {
    jv a = A();
    jv b = B();
    c = function(a, b);
  }
```

In this case, you have:
```c
  paths = jv_sort(jv_array_slice(paths, 0, jv_array_length(jv_copy(paths))), jv_copy(paths));
```

If the compiler generates code that runs this as:
```c
  {
    jv copy = jv_copy(paths);
    jv slice = jv_array_slice(paths, 0, jv_array_length(jv_copy(paths)));
    paths = jv_sort(slice, copy);
  }
```
Then it is fine because you are not using a value that has already been freed anywhere.

But if the compiler generates code that runs it is as:
```c
  {
    jv slice = jv_array_slice(paths, 0, jv_array_length(jv_copy(paths)));
    jv copy = jv_copy(paths);
    paths = jv_sort(slice, copy);
  }
```
Then it is not fine because `jv_array_slice(paths,...)` will have freed `paths`, and `jv_copy(paths)` will try to copy `paths` after it has been freed.

As a further example, the problem could also be fixed using:
```c
  jv copy = jv_copy(paths);
  paths = jv_sort(jv_array_slice(paths, 0, jv_array_length(jv_copy(paths))), copy);
```